src/components/queryBuilder/OrderBy.tsx

```diff
@@ -171,10 +174,18 @@ export const getOrderByFields = (
           return { value: g, label: g };
         }),
       ];
+      break;
     case BuilderMode.List:
     default:
-      return fieldsList.map((m) => {
+      values = fieldsList.map((m) => {
         return { value: m.name, label: m.label };
       });
   }
+  // Add selected value to the list if it does not exist.
+  (builder as SqlBuilderOptionsAggregate).orderBy?.forEach((x) => {
```

Comment:
again.  .filter may be a bit cleaner.  any time you have an if condition within a for loop think of using .filter